### ISSUE TITLE
Can't get to create labs - validate hangs not finding get-ad commands

### ISSUE THREAD

**Issue by ahinkle1965** [pluralsight/PS-AutoLab-Env#153]
Please help!  I've been messing with this for two solid days nows.  I've tried under win10 and win server 2016.  I can get the lab TEST-SingleServer to complete, so I think i can say the package is working.  But when I try to do the POC-MultiRole it never completes.  I am very fluent in VMWorkstation, ESX, and Vcenter, but never ran anything under Hyper-v.  The validate ran over night and still issues errors:

The term 'Get-ADDomain' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
the spelling of the name, or if a path was included, verify that the path is correct and try again.
    + CategoryInfo          : ObjectNotFound: (Get-ADDomain:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
    + PSComputerName        : DC1

The term 'Get-ADorganizationalUnit' is not recognized as the name of a cmdlet, function, script file, or operable
program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
    + CategoryInfo          : ObjectNotFound: (Get-ADorganizationalUnit:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
    + PSComputerName        : DC1

The term 'Get-ADGroup' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
the spelling of the name, or if a path was included, verify that the path is correct and try again.
    + CategoryInfo          : ObjectNotFound: (Get-ADGroup:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
    + PSComputerName        : DC1

The term 'Get-AdUser' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
    + CategoryInfo          : ObjectNotFound: (Get-AdUser:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
    + PSComputerName        : DC1

The term 'Get-ADComputer' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
the spelling of the name, or if a path was included, verify that the path is correct and try again.
    + CategoryInfo          : ObjectNotFound: (Get-ADComputer:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
    + PSComputerName        : DC1

I added AD to my systems as any of these commands failed when entered in just a PS console.  But they at least attempt to run now:

PS C:\AutoLab\Configurations\POC-MultiRole> Get-ADComputer
cmdlet Get-ADComputer at command pipeline position 1
Supply values for the following parameters:
(Type !? for Help.)
Filter:  

Also, I cannot open a session into the DC1 system, it just sits and spins....

**Comment by ahinkle1965** [comment]
and, oh, i can program and script pretty well.  But never used powershell.  c, c++, rexx, bash, korn, etc.  not very fluent in AD stuff either...

**Comment by ahinkle1965** [comment]
Thanks.  I'll look at that. 
I let it run overnight and it never finished. 


Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Jeff Hicks <notifications@github.com> Date: 7/6/18  9:27 AM  (GMT-05:00) To: theJasonHelmick/PS-AutoLab-Env <PS-AutoLab-Env@noreply.github.com> Cc: ahinkle1965 <ahinkle65@gmail.com>, Author <author@noreply.github.com> Subject: Re: [theJasonHelmick/PS-AutoLab-Env] Can't get to create labs - validate hangs not finding get-ad commands (#153) 
When the validation tests first runs you will see errors like that. But eventually everything should resolve. Some of those configurations are a bit old and a lot has changed. Make sure you are running the latest version of Autolab. You might also try my PowerShell Lab configuration. https://github.com/jdhitsolutions/PowerShellLab

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/theJasonHelmick/PS-AutoLab-Env","title":"theJasonHelmick/PS-AutoLab-Env","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/theJasonHelmick/PS-AutoLab-Env"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jdhitsolutions in #153: When the validation tests first runs you will see errors like that. But eventually everything should resolve. Some of those configurations are a bit old and a lot has changed. Make sure you are running the latest version of Autolab. You might also try my PowerShell Lab configuration. https://github.com/jdhitsolutions/PowerShellLab"}],"action":{"name":"View Issue","url":"https://github.com/theJasonHelmick/PS-AutoLab-Env/issues/153#issuecomment-403034019"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/theJasonHelmick/PS-AutoLab-Env/issues/153#issuecomment-403034019",
"url": "https://github.com/theJasonHelmick/PS-AutoLab-Env/issues/153#issuecomment-403034019",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [theJasonHelmick/PS-AutoLab-Env] Can't get to create labs - validate hangs not finding get-ad commands (#153)",
"sections": [
{
"text": "",
"activityTitle": "**Jeff Hicks**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@jdhitsolutions",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"theJasonHelmick/PS-AutoLab-Env\",\n\"issueId\": 153,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"theJasonHelmick/PS-AutoLab-Env\",\n\"issueId\": 153\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/theJasonHelmick/PS-AutoLab-Env/issues/153#issuecomment-403034019"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 353568608\n}"
}
],
"themeColor": "26292E"
}
]

**Comment by ahinkle1965** [comment]
Hey there.

 

I pulled PowerShell lab and ran it.  Pretty much the same results.  After two hours, and still getting the same errors.  I try to connect and it sits and spins.  One thing I noticed additional though, only srv3 has a 192.168 address.

 

The rest have a 169.254.  So. . .  to me that suggests that none of the config is getting ran anywhere except in srv3.  I don’t know where to look in the clients to see the scripts.  Was thinking if I knew that, I could try them by hand.  I can connect to a guest only after I reboot it though.

 

I am running this as a VM (VM Workstation) and I do have it set as hypervisor.  I saw in either your notes or the plural site notes you had to set the host win10/win2016 that way.  It doesn’t make sense though, that srv3 config’d and the others didn’t.  I am attempting to work thru the plural course on pki.

 

Oh yeah, I saw in one of the issues on git hub, somebody had to install some power shell modules and modify the version in a definition file.  I did that too, initially, but I later found that adding the –requiredversion option to the install module command brought down the correct versions.  If I stumble across that issue, I’ll update it, if you think that’s a better solution.  Once again, my terminology may not be right I haven’t done much in the way of powershell.  I’ve been buried under AIX and vmware esx the last 10 years or so.

**Comment by tj76352** [comment]
Been getting the same errors on validation...it has been attempting to validate for 3 days now. Any ideas?

**Comment by ahinkle1965** [comment]
Hi.  Sorry it has taken me a while.  I looked back thru notes.  In my case, I was running VM's (Workstation).  The only way I got it to work was fresh/clean windows install into a VM, and as little tweaks as possible.  I seem to remember multiple lab buildouts I was working with or trying.  Do the build out if everything in one pass (IE don't build one lab machine, then build others in a second script run).  Think of it this way. . . you get one build run per windows installation.  The first works, rebuilding or building additional never passed validation.  

I hope this helps, it was quite a while ago (at least for me).

**Comment by jdhitsolutions** [comment]
This project is now under new ownership. The 3.x version line is obsolete. Version 4.0.0 has been published to the PowerShell Gallery with new updates, configuration, and documentation. It is expected that this version will resolve most if not all of the problems you were experiencing. Read [Updating](https://github.com/pluralsight/PS-AutoLab-Env/blob/master/Updating.md) for more information.

No further action will be taken on this issue and it will most likely be closed at some point. If you still have the same issue with the new version, then go ahead and open a new issue.
